lib/python/picongpu/plugins/plot_mpl/emittance_evolution_visualizer.py

```diff
@@ -0,0 +1,158 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat
+License: GPLv3+
+"""
+
+from picongpu.plugins.emittance import Emittance
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer, plt
+import numpy as np
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of histogram plots on a logscaled y-axis.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super(Visualizer, self).__init__(run_directory)
+        self.cbar = None
+
+    def _create_data_reader(self, run_directory):
+        """
+        Implementation of base class function.
+        """
+        return Emittance(run_directory)
+
+    def _create_plt_obj(self, ax):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        emit, y_slices, iteration = self.data
+        np_data = np.zeros(len(iteration))
+        for index, ts in enumerate(iteration):
+            np_data[index] = emit[ts][0]
+        self.plt_obj, = ax.plot(iteration * 1.39e-16 * 1.e12,
```

Comment:
the magic numbers here need explanation, ideally by storing them in a temporary, well named variable.
Looks like you need `dt`, so you can use the `FindTime` module in `python/util/` that we have.


Why is once `self.itera` and once `iteration` used? They are named very similar but are they the same?
I would store `ps = 1.e12` (and what does the `1.e6` scale to?) as variables to make things easier readable.

Reply:
`self.itera` and `iteration ` are not the same.` self.itera` is the currently selected iteration step (renamed to `self.iteration`), `iteration` contains all iteration steps (renamed to `all_iterations`)